fdbserver/DDTeamCollection.actor.cpp

```diff
@@ -1951,56 +1951,34 @@ class DDTeamCollectionImpl {
-		int count = 0;
-		for (auto& [id, s] : self->server_info) {
-			// If a healthy SS don't have storage metrics, skip this round
-			if (self->server_status.get(s->getId()).isUnhealthy()) {
-				continue;
-			} else if (!s->metricsPresent()) {
-				return 0;
-			}
-
-			double load = s->loadBytes();
-			avgLoadBytes += load;
-			++count;
-			minLoadBytes = std::min(minLoadBytes, load);
-		}
-		avgLoadBytes /= count;
-		return minLoadBytes / avgLoadBytes;
-	}
-
 	ACTOR static Future<Void> perpetualStorageWiggleRest(DDTeamCollection* self) {
-		state Future<Void> maxDelay = delay(SERVER_KNOBS->PERPETUAL_WIGGLE_DELAY);
+		wait(delay(SERVER_KNOBS->PERPETUAL_WIGGLE_DELAY));
 		state bool takeRest = true;
 		while (takeRest) {
-			choose {
-				when(wait(maxDelay)) { break; }
-				when(double balance = wait(loadBytesBalanceRatio(self))) {
-					// there must not have other teams to place wiggled data
-					takeRest = self->server_info.size() <= self->configuration.storageTeamSize ||
-					           self->machine_info.size() < self->configuration.storageTeamSize ||
-					           balance < SERVER_KNOBS->PERPETUAL_WIGGLE_MIN_BYTES_BALANCE;
-
-					CODE_PROBE(balance < SERVER_KNOBS->PERPETUAL_WIGGLE_MIN_BYTES_BALANCE,
-					           "Perpetual Wiggle pause because cluster is imbalance.");
-
-					if (takeRest) {
-						self->storageWiggler->setWiggleState(StorageWiggler::PAUSE);
-						if (self->configuration.storageMigrationType == StorageMigrationType::GRADUAL) {
-							TraceEvent(SevWarn, "PerpetualStorageWiggleSleep", self->distributorId)
-							    .suppressFor(SERVER_KNOBS->PERPETUAL_WIGGLE_DELAY * 4)
-							    .detail("BytesBalance", balance)
-							    .detail("ServerSize", self->server_info.size())
-							    .detail("MachineSize", self->machine_info.size())
-							    .detail("StorageTeamSize", self->configuration.storageTeamSize);
-						}
-					}
+			bool balance = self->loadBytesBalanceRatio();
+			// there must not have other teams to place wiggled data
+			takeRest = self->server_info.size() <= self->configuration.storageTeamSize ||
+			           self->machine_info.size() < self->configuration.storageTeamSize ||
+			           balance < SERVER_KNOBS->PERPETUAL_WIGGLE_MIN_BYTES_BALANCE_RATIO;
+
+			if (takeRest) {
+				CODE_PROBE(balance < SERVER_KNOBS->PERPETUAL_WIGGLE_MIN_BYTES_BALANCE_RATIO,
+				           "Perpetual Wiggle pause because cluster is imbalance.");
+
+				self->storageWiggler->setWiggleState(StorageWiggler::PAUSE);
+				if (self->configuration.storageMigrationType == StorageMigrationType::GRADUAL) {
+					TraceEvent(SevWarn, "PerpetualStorageWiggleSleep", self->distributorId)
+					    .suppressFor(SERVER_KNOBS->PERPETUAL_WIGGLE_DELAY * 4)
+					    .detail("BytesBalance", balance)
+					    .detail("ServerSize", self->server_info.size())
+					    .detail("MachineSize", self->machine_info.size())
+					    .detail("StorageTeamSize", self->configuration.storageTeamSize);
 				}
+			} else {
+				break;
```

Comment:
Optional simplification here that would not change logic:

Because of this `break`, you would exit the loop any time `takeRest` is set to `false`, so you don't actually need `takeRest` to be a state variable outside of the loop, and the `while(takeRest)` can just be `loop` since the `break` is required to avoid the timeout if it is time to exit.